code/hpke/Hacl.Impl.HPKE.fst

```diff
@@ -0,0 +1,492 @@
+    (S.id_of_mode S.Base `Seq.append`
+    S.id_of_cs cs `Seq.append`
+    as_seq h0 pkE `Seq.append`
+    as_seq h0 pkR `Seq.append`
+    as_seq h0 pkI `Seq.append`
+    Lib.ByteSequence.nat_to_bytes_be 1 hlength `Seq.append`
+    as_seq h0 pskID_hash `Seq.append`
+    Lib.ByteSequence.nat_to_bytes_be 1 hlength `Seq.append`
+    as_seq h0 info_hash))
+
+noextract inline_for_extraction
+val ks_derive_default_aux:
+     #cs:S.ciphersuite
+  -> pkR:key_dh_public cs
+  -> zz:key_dh_public cs
+  -> pkE:key_dh_public cs
+  -> infolen: size_t{v infolen <= S.max_info}
+  -> info: lbuffer uint8 infolen
+  -> o_key: key_aead cs
+  -> o_nonce: nonce_aead cs
+  -> context_len:size_t{v context_len == 9 + (3 * S.size_dh_public cs) + 2 *
+                                     Spec.Agile.Hash.size_hash (S.hash_of_cs cs)}
+  -> context:lbuffer uint8 context_len
+  -> secret:lbuffer uint8 (nhash_length cs)
+  -> pkI:lbuffer uint8 (nsize_dh_public cs)
+  -> psk:lbuffer uint8 (nhash_length cs)
+  -> label_key:lbuffer uint8 8ul
+  -> label_nonce:lbuffer uint8 10ul
+  -> tmp:lbuffer uint8 (10ul +. context_len)
+  -> ST unit
+       (requires fun h0 ->
+         live h0 pkR /\ live h0 zz /\ live h0 pkE /\
+         live h0 info /\ live h0 o_key /\ live h0 o_nonce /\
+         live h0 context /\ live h0 secret /\ live h0 pkI /\
+         live h0 psk /\ live h0 tmp /\ live h0 label_key /\ live h0 label_nonce /\
+         disjoint o_key o_nonce /\ disjoint secret psk /\ disjoint secret zz /\
+         disjoint context zz /\ disjoint context psk /\ disjoint context secret /\
+         disjoint tmp label_key /\ disjoint tmp label_nonce /\ disjoint tmp context /\
+         disjoint label_key context /\ disjoint label_key secret /\
+         disjoint label_nonce o_key /\ disjoint label_nonce context /\
+         disjoint context pkE /\ disjoint context pkR /\ disjoint context pkI /\ disjoint context info /\
+         disjoint label_nonce secret /\
+         disjoint tmp zz /\ disjoint tmp psk /\
+         disjoint tmp pkE /\ disjoint tmp pkI /\ disjoint tmp pkR /\
+         disjoint secret tmp /\ disjoint o_key tmp /\ disjoint tmp info /\
+         disjoint o_key secret /\ disjoint o_nonce secret /\
+
+         as_seq h0 label_key `Seq.equal` S.label_key /\
+         as_seq h0 label_nonce `Seq.equal` S.label_nonce /\
+         as_seq h0 psk `Seq.equal` S.default_psk cs /\
+         as_seq h0 pkI `Seq.equal` S.default_pkI cs)
+       (ensures fun h0 _ h1 -> modifies (loc o_nonce |+| loc o_key |+| loc tmp |+| loc context |+| loc secret) h0 h1 /\
+         (let keyIR, nonceIR = S.ks_derive cs S.Base (as_seq h0 pkR) (as_seq h0 zz) (as_seq h0 pkE) (as_seq h0 info) None None in
+         as_seq h1 o_key == keyIR /\ as_seq h1 o_nonce == nonceIR))
+
+#set-options "--z3rlimit 100"
+
+noextract inline_for_extraction
```

Comment:
you should not need to write `inline_for_extraction` yourself here -- the tactic will automatically add it, and this may speed up extraction time for this file (since the normalizer won't have as much work to do)

Reply:
Good catch, thanks. I removed the `inline_for_extraction` annotation from all functions with the inline_ tactic attribute